packages/variable-resolver/src/browser/variable-resolver-service.ts

```diff
@@ -31,20 +31,24 @@ export class VariableResolverService {
 
     /**
      * Resolve the variables in the given string array.
+     * @param value The array of data to resolve
+     * @param registry The variable registry to use in the variable resolution
      * @returns promise resolved to the provided string array with already resolved variables.
      * Never reject.
      */
-    resolveArray(value: string[]): Promise<string[]> {
-        return this.resolve(value);
+    resolveArray(value: string[], registry: VariableRegistry = this.variableRegistry): Promise<string[]> {
```

Comment:
why do we pass `VariableRegistry` here? clients should use `VariableContribution` to contribute variables